receivers/opsgenie/config.go

```diff
@@ -74,6 +86,33 @@ func NewConfig(jsonData json.RawMessage, decryptFn receivers.DecryptFunc) (Confi
 		raw.OverridePriority = &overridePriority
 	}
 
+	for idx, r := range raw.Responders {
+		if r.ID == "" && r.Username == "" && r.Name == "" {
+			return Config{}, fmt.Errorf("responder at index [%d] must have at least one of id, username or name specified", idx)
+		}
+		if strings.Contains(r.Type, "{{") {
+			_, err := template.New("").Parse(r.Type)
+			if err != nil {
+				return Config{}, fmt.Errorf("responder at index [%d] type is not a valid template: %v", idx, err)
+			}
+		} else {
```

Comment:
But why not remove the `else` and let the template be checked as any other value after it's evaluated?

Reply:
because the result of the template expansion could be different when it is applied with actual alerts as the context. For example `{{ .CommonAnnotations.ResponderType }}`

Reply:
Please see the test https://github.com/grafana/alerting/pull/131/files#diff-0c8a173e84ce8f8f978f1403db2f1687d0cf4aa8ef8428fe5bda9607c86f5a5aR268-R271

Reply:
Ah, makes sense!